core/include/traccc/clusterization/impl/measurement_creation.ipp

```diff
@@ -92,38 +99,53 @@ TRACCC_HOST_DEVICE inline void fill_measurement(
     // [2] The Art of Computer Programming, Donald E. Knuth, second
     //     edition, chapter 4.2.2.
 
-    // A security check.
-    assert(cluster.empty() == false);
+    // Security checks.
+    assert(clusters.cell_indices()[index].empty() == false);
+    assert([&]() {
+        const unsigned int module_idx =
+            cells.module_index().at(clusters.cell_indices()[index].front());
+        for (const unsigned int cell_idx : clusters.cell_indices()[index]) {
+            if (cells.module_index().at(cell_idx) != module_idx) {
+                return false;
+            }
+        }
+        return true;
+    }() == true);
 
     // Calculate the cluster properties
     scalar totalWeight = 0.f;
     point2 mean{0.f, 0.f}, var{0.f, 0.f};
-    calc_cluster_properties(cluster, det_descr, mean, var, totalWeight);
+    calc_cluster_properties(index, cells, clusters, det_descr, mean, var,
+                            totalWeight);
```

Comment:
Does this still need to take an index now that we have proxies?

Reply:
See: https://github.com/acts-project/vecmem/pull/298 Proxying clusters doesn't work with vecmem `v1.9.0`. :frowning:

Reply:
Done.